internal/provider/resource_bootstrap_git.go

```diff
@@ -649,10 +656,16 @@ func (r bootstrapGitResource) Delete(ctx context.Context, req resource.DeleteReq
 		resp.Diagnostics.AddError("Unable to remove CRDs", err.Error())
 		return
 	}
-	err = uninstall.Namespace(ctx, log.NopLogger{}, kubeClient, data.Namespace.ValueString(), false)
-	if err != nil {
-		resp.Diagnostics.AddError("Unable to remove namespace", err.Error())
-		return
+
+	// Only remove namespace if not keeping it.
+	if data.KeepNamespace.ValueBool() {
+		tflog.Debug(ctx, fmt.Sprintf("The keep_namespace variable was set to true. Skipping removal of %s namespace.", data.Namespace), map[string]interface{}{})
+	} else {
+		err = uninstall.Namespace(ctx, log.NopLogger{}, kubeClient, data.Namespace.ValueString(), false)
+		if err != nil {
+			resp.Diagnostics.AddError("Unable to remove namespace", err.Error())
+			return
```

Comment:
We need to remove the `return` from here and all the above functions. This is what we do in the CLI, maybe the controllers were deleted by hand, or the CRDs, or even the namespace. We should log the not found errors and continue with the cleanup.